### PR TITLE
7903270: Util classes should be final and have a private constructor

### PR DESCRIPTION
**Pull request by minborg**
This PR makes some of the classes and fields `final`. Also, a general cleanup of the `RuntimeHelper` class is proposed.

There are some files that are "overhidden" (by attempting to "override"/hide static methods). This will be fixed in a separate PR and so these classes are not affected in this PR.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed (no review required)

### Issue
 * [CODETOOLS-7903270](https://bugs.openjdk.org/browse/CODETOOLS-7903270): Util classes should be final and have a private constructor

### Reviewers
 * [Athijegannathan Sundararajan](https://openjdk.org/census#sundar) (@sundararajana - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jextract pull/65/head:pull/65` \
`$ git checkout pull/65`

Update a local copy of the PR: \
`$ git checkout pull/65` \
`$ git pull https://git.openjdk.org/jextract pull/65/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 65`

View PR using the GUI difftool: \
`$ git pr show -t 65`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jextract/pull/65.diff">https://git.openjdk.org/jextract/pull/65.diff</a>

</details>

### CODE REVIEW

File: src/main/java/org/openjdk/jextract/impl/ClassSourceBuilder.java [hunk 1]
```diff
@@ -174,7 +198,13 @@ boolean isEnclosedBySameName(String name) {
     protected void emitPackagePrefix() {
         if (!isNested()) {
             assert packageName().indexOf('/') == -1 : "package name invalid: " + packageName();
-            append("// Generated by jextract\n\n");
+            append("// Generated by jextract");
+            if (SHOW_GENERATING_CLASS) {
```
Comment (minborg) [inline src/main/java/org/openjdk/jextract/impl/ClassSourceBuilder.java:202]: This is very useful when figuring out how a class is generated.

**Review by sundararajana — APPROVED** [review]
LGTM

**Pull request closed without merging**